server/server.go

```diff
@@ -179,24 +180,35 @@ func (s *server) buildInvokeChain() {
 
 // RegisterService should not be called by users directly.
 func (s *server) RegisterService(svcInfo *serviceinfo.ServiceInfo, handler interface{}) error {
-	if s.svcInfo != nil {
-		panic(fmt.Sprintf("Service[%s] is already defined", s.svcInfo.ServiceName))
+	if s.isStarted {
```

Comment:
The whole RegisterService should add the lock, and rename `isStarted` to `isRun` is better, because we set `isRun` to true once we call Run()